docs/reference/db-config.rst

```diff
@@ -1,8 +1,41 @@
 :orphan:
 
-=======================
-db/config.cc properties
-=======================
+======================
+db/config.cc reference
+======================
 
-.. datatemplate:yaml:: ../_data/db_config.yaml
+Configuration properties
+------------------------
+
+.. dbconfigdatatemplate:: ../_data/db_config.yaml
+  :template: db_config.tmpl
+  :filter: Used
+
+Deprecated properties
```

Comment:
To do: Remove deprecated properties

Reply:
@dgarcia360 hi David, in case you are still interested. we are trying to document the options in code as well. see https://github.com/scylladb/scylladb/pull/15277 . and the original author of these states enum explained the intention at https://github.com/scylladb/scylladb/pull/15277#discussion_r1317660992 .